addons/website_event/views/event_templates_page.xml

```diff
@@ -6,65 +6,57 @@
-                            <span class="navbar-toggler-icon"></span>
-                        </button>
-                        <div id="o_wevent_event_submenu" class="collapse navbar-collapse">
-                            <ul class="navbar-nav w-100 flex-md-wrap" t-att-data-menu_name="editable and 'Event Menu'" t-att-data-content_menu_id="editable and event.menu_id.id">
-                                <t t-foreach="event.menu_id.child_id" t-as="submenu">
-                                    <t t-call="website.submenu">
-                                        <t t-set="item_class" t-value="'nav-item'"/>
-                                        <t t-set="link_class" t-value="'nav-link'"/>
-                                    </t>
-                                </t>
-                            </ul>
-                            <!-- Add Register additional CTA button, in addition to menus -->
-                            <a t-if="event.menu_register_cta and not event.is_participating"
-                                t-att-href="'/event/%s/register' % (slug(event))"
-                                class="btn btn-primary ms-auto">
-                                Register
-                            </a>
-                        </div>
-                    </div>
-                </nav>
-            </t>
+        <div id="wrap" t-attf-class="o_wevent_event js_event d-flex flex-column h-100 #{'o_wevent_hide_sponsors' if hide_sponsors else ''}">
+            <t t-if="not hide_navbar" t-call="website_event.navbar"/>
             <t t-out="0"/>
             <t t-set="editor_sub_message">Following content will appear on all events.</t>
             <div class="oe_structure oe_empty" id="oe_structure_website_event_layout_1" t-att-data-editor-sub-message="editor_sub_message"/>
         </div>
     </t>
 </template>
 
+<template id="navbar" name="Event Navbar">
+    <div t-attf-class="container #{'my-3' if not event.menu_id else 'mt-3'}">
+        <a href="/event">
+            <i class="fa fa-chevron-left me-2"/>
+            <span>Back to events</span>
+        </a>
+        <ul class="navbar-nav flex-row ms-md-auto ms-0">
+            <li t-if="opt_events_list_categories" class="nav-item me-3">
+                <a t-attf-href="/event?type=#{event.event_type_id.id}" t-if="event.event_type_id" class="nav-link">
+                    <i class="fa fa-folder-open me-2"/><span t-field="event.event_type_id"/>
+                </a>
+            </li>
+        </ul>
+    </div>
+    <div class="mb-3">
+        <div class="container d-flex justify-content-between align-items-center">
+            <h6 t-field="event.name" class="flex-grow-1 mb-0"/>
+            <!-- Add Register additional CTA button, in addition to menus -->
+            <a t-if="event.menu_register_cta and not event.is_participating"
```

Comment:
To hide this when on the '/register' page, you can just add another condition. This time I would actually do it from the route like we originally did for the others.

So here add `and not hide_register_cta` and in the `def _prepare_event_register_values` add an entry `'hide_register_cta': True`